accord-core/src/main/java/accord/messages/Apply.java

```diff
@@ -54,15 +63,59 @@ public Apply(Node.Id to, Topologies topologies, TxnId txnId, Txn txn, Key homeKe
         this.result = result;
     }
 
+    @VisibleForImplementation
+    public Apply(Keys scope, long waitForEpoch, TxnId txnId, Txn txn, Key homeKey, Timestamp executeAt, Deps deps, Writes writes, Result result)
+    {
+        super(scope, waitForEpoch);
+        this.txnId = txnId;
+        this.txn = txn;
+        this.homeKey = homeKey;
+        this.executeAt = executeAt;
+        this.deps = deps;
+        this.writes = writes;
+        this.result = result;
+    }
+
+    static Future<Void> waitAndReduce(Future<Void> left, Future<Void> right)
+    {
+        try
+        {
+            if (left != null) left.get();
+            if (right != null) right.get();
+        }
+        catch (InterruptedException e)
+        {
+            throw new UncheckedInterruptedException(e);
```

Comment:
In testing accord.messages.Apply#waitAndReduce swallows the InterruptedException, causing Instance.shutdown to hang. This should re-add the interrupt to the thread